django_lightweight_queue/management/commands/queue_runner.py

```diff
@@ -1,62 +1,81 @@
-            help="Total number of machines running the queues"),
-        optparse.make_option('--only-queue', action='store', default=None,
-            help="Only run the given queue, useful for local debugging"),
-        optparse.make_option('--num-queue-workers', action='store', default=None,
-            help="The number of queue workers to run, only valid when running a specific queue"),
-    )
-
-    def handle_noargs(self, **options):
-        # Django < 1.8.3 leaves options['verbosity'] as a string so we cast to
-        # ensure an int.
-        verbosity = int(options['verbosity'])
 
-        if (
-            options['num_queue_workers'] is not None and
-            options['only_queue'] is None
-        ):
-            raise CommandError(
-                "A value for 'queue-workers' without a value for 'only-queue' "
-                "has no meaning.",
-            )
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('--pidfile', action='store', dest='pidfile', default=None,
+            help="Fork and write pidfile to this file.")
+        parser.add_argument('--logfile', action='store', dest='logfile', default=None,
+            help="Log to the specified file.")
+        parser.add_argument('--touchfile', action='store', dest='touchfile', default=None,
+            help="touch(1) the specified file after running a job.")
+        parser.add_argument('--machine', action='store', dest='machine_number', default=None,
+            help="Machine number, for parallelism")
+        parser.add_argument('--of', action='store', dest='machine_count', default=None,
+            help="Total number of machines running the queues")
+        parser.add_argument('--only-queue', action='store', default=None,
+            help="Only run the given queue, useful for local debugging")
+        parser.add_argument('--config', action='store', default=None,
+            help="The path to an additional django-style config file to load")
+        parser.add_argument('--exact-configuration', action='store_true',
+            help="Run queues on this machine exactly as specified. Requires the"
+                 " use of the '--config' option in addition. It is an error to"
+                 " use this option together with either '--machine' or '--of'.")
+
+    def validate_and_normalise(self, options):
+        if options['exact_configuration']:
+            if not options['config']:
+                raise CommandError(
+                    "Must provide a value for '--config' when using "
+                    "'--exact-configuration'.",
+                )
+
+            invalid_others = ('machine_count', 'machine_number', 'only_queue')
+
+            if any(options[name] is not None for name in invalid_others):
+                raise CommandError(
+                    "Must not specify '--machine', '--of' or '--only-queue'"
+                    " when using '--exact-configuration'.",
+                )
```

Comment:
Not sure what you mean? The main thing we're doing here is detecting invalid combinations of arguments; I'm not aware of a way to do that directly in either argparse or Django. I'd love to use a built-in mechanism if there is one!